typescript/core/src/McapStreamReader.ts

```diff
@@ -89,7 +91,41 @@ export default class McapStreamReader {
     if (this.#doneReading) {
       throw new Error("Already done reading");
     }
-    this.#buffer.append(data);
+    this.#appendOrShift(data);
+    this.#reader.reset(this.#view);
+  }
+  #appendOrShift(data: Uint8Array): void {
```

Comment:
nit: append() is so small I could see merging appendOrShift into it as a single function. Initially I was confused why `#reader` was not updated in appendOrShift, then I noticed it was here – that update could just be moved to the end of the function if they are combined